### PR TITLE
Add in gini and information gain split criteria

### PR DESCRIPTION
**Pull request by hectorgon**
@deerzq @tension7

### CODE REVIEW

File: training/src/main/scala/com/airbnb/aerosolve/training/DecisionTreeTrainer.scala [hunk 1]
```diff
@@ -156,14 +162,53 @@ object DecisionTreeTrainer {
...
+          val n1 = rightNeg / rightCount
+          val f1 = rightCount / (leftCount + rightCount)
+          val p2 = leftPos / leftCount
+          val n2 = leftNeg / leftCount
+          val f2 = leftCount / (leftCount + rightCount)
+          splitCriteria match {
+            case "gini" => {
+              // Using negative gini since we are maximizing.
+              val gini = - (f1 * (p1 * (1.0 - p1)  + n1 * (1.0 - n1)) +
+                  f2 * (n2 * (1.0 - n2) + p2 * (1 - p2)))
```
Comment (spencerde) [inline training/src/main/scala/com/airbnb/aerosolve/training/DecisionTreeTrainer.scala]: super nitpick `1.0 - p2` for consistency

**Comment by spencerde** [comment]
Nice LGTM.  

The unit tests don't actually differentiate between the different split methods it just tests that those strings are accepted.  If there was a clever way to easily get to a different fraction correct w/ the different split criteria that would be even better but it's probably fine.

**Comment by hectorgon** [comment]
The unit tests just check the outcome yeah doesn't actually test the formula. The fractionCorrect sort of varies from run to run so I use a conservative lower bound.

**Comment by spencerde** [comment]
Checking the outcome is still fairly good protection I see what you are saying :+1:

**Pull request merged**